heareval/predictions/task_predictions.py

```diff
@@ -781,20 +866,29 @@ def print_scores(mode, scores_and_trainers):
             gpus=gpus,
             deterministic=deterministic,
         )
-        scores_and_trainers.append((best_model_score, trainer, predictor))
+        scores_and_trainers.append(
+            (best_model_score, model_path, epoch, trainer, hparams)
+        )
         print_scores(mode, scores_and_trainers)
 
     with open(embedding_path.joinpath("valid.grid.jsonl"), "wt") as w:
-        for score, _, predictor in scores_and_trainers:
-            w.write(json.dumps([score, hparams_to_json(predictor.hparams)]) + "\n")
+        for score, _, _, _, predictor in scores_and_trainers:
+            w.write(json.dumps([score, hparams_to_json(hparams)]) + "\n")
 
     # Use that model to compute test scores
-    best_score, best_trainer, best_predictor = scores_and_trainers[0]
+    (
+        best_score,
+        best_model_path,
+        best_model_epoch,
+        best_trainer,
+        best_hparams,
+    ) = scores_and_trainers[0]
```

Comment:
And so maybe we are picking only the last random `hparam` in the config where we ended in the `for loop` for the hyperparameter selection